### DISCUSSION TITLE
OpenID timeout (auth)

### DISCUSSION THREAD

**Discussion by rjcpereira** [nuxt/nuxt#17916] [Questions] [answered]
There is any way to use the [express-openid-connect](https://auth0.github.io/express-openid-connect/index.html) module for authentication?

I've a **auth.js** file in **/server/middleware**:

```javascript
import express from 'express';
import { auth } from 'express-openid-connect';
import cookieParser from 'cookie-parser';

const app = express();

const config = {
    authRequired: true,
    issuerBaseURL: process.env.OPEN_ID_URL,
    baseURL: process.env.OPEN_BASE_URL,
    clientID: process.env.OPEN_ID_CLIENT,
    secret: process.env.OPEN_ID_SECRET,
    idpLogout: true,
    authorizationParams: {
        _scheme: 'oauth2',
        authorization_endpoint: `${process.env.OPEN_ID_URL}/authorize`,
        redirect_uri: `${process.env.OPEN_BASE_URL}/auth/callback`,
        scope: 'openid role email profile corporate_username',
        response_type: 'id_token',
        token_key: 'access_token',
        token_type: 'Bearer'
    }
};

app.use(cookieParser());
app.use(express.json({limit: '5mb'}));
app.use(express.urlencoded({limit: '5mb'}));

app.use(auth(config));

app.use((req, res, next) => {
    console.warn('OPEN ID', config, req.oidc)
    next()
});

export default app
```

If the authentication is required (config), it throws an error and don't return nothing:

> ERROR  [worker] Issuer.discover() failed.
>    RequestError: Timeout awaiting 'request' for 5000ms
>    RequestError: Timeout awaiting 'request' for 5000ms
>  RequestError: Timeout awaiting 'request' for 5000ms
>  RequestError: Timeout awaiting 'request' for 5000ms
>  at Function.discover (node_modules/openid-client/lib/issuer.js:265:17)
>  at processTicksAndRejections (node:internal/process/task_queues:96:5)
>  at async pReflect (node_modules/p-reflect/index.js:5:17)
>  at async Promise.all (index 0)
>  at async node_modules/p-memoize/index.js:33:22
>  at async get (node_modules/express-openid-connect/lib/client.js:43:18)
>  at async pReflect (node_modules/p-reflect/index.js:5:17)
>  at async Promise.all (index 0)
>  at async get (node_modules/p-memoize/index.js:33:22)
>  at async ResponseContext.login (node_modules/express-openid-connect/lib/context.js:182:20)

> ERROR  [proxy] socket hang up
>  at connResetException (node:internal/errors:691:14)
>  at Socket.socketOnEnd (node:_http_client:466:23)
>  at Socket.emit (node:events:532:35)
>  at Socket.emit (node:domain:475:12)
>  at endReadableNT (node:internal/streams/readable:1346:12)
>  at processTicksAndRejections (node:internal/process/task_queues:83:21)

**Comment by wobsoriano** [comment, marked as answer]
Use [Runtime Config](https://v3.nuxtjs.org/docs/usage/runtime-config) to access your credentials

```ts
// nuxt.config.ts
import { defineNuxtConfig } from "nuxt3";

// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config
export default defineNuxtConfig({
  privateRuntimeConfig: {
    OPEN_ID_CLIENT: process.env.OPEN_ID_CLIENT,
    OPEN_ID_SECRET: process.env.OPEN_ID_SECRET,
  },
  publicRuntimeConfig: {
    OPEN_BASE_URL: process.env.OPEN_BASE_URL,
    OPEN_ISSUER_BASE_URL: process.env.OPEN_ISSUER_BASE_URL,
  },
});
```

```ts
// server/middleware/auth0.ts
import express from "express";
import { auth } from "express-openid-connect";
import cookieParser from "cookie-parser";
import runtimeConfig from "#config";

const app = express();

const config = {
  issuerBaseURL: runtimeConfig.OPEN_ISSUER_BASE_URL,
  baseURL: runtimeConfig.OPEN_BASE_URL,
  clientID: runtimeConfig.OPEN_ID_CLIENT,
  secret: runtimeConfig.OPEN_ID_SECRET,
  idpLogout: true,
};

app.use(cookieParser());
app.use(express.json({ limit: "5mb" }));
app.use(express.urlencoded({ limit: "5mb" }));

app.use(auth(config));

export default app;
```

```ts
// server/api/user.ts
export default async (req, res) => {
  if (!req.oidc.isAuthenticated()) {
    res.statusCode = 401;
    return res.end("Please login");
  }

  return req.oidc.user;
};
```

**Reply by rjcpereira** [reply]
Sorry, that wasn't the issue. Trying this with the company VPN disconnected was. 
Anyway, changed all of my process.env to runtimeConfig, thanks
